.githooks/pre-commit

```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
+
+tmp=$(mktemp)
+git show :config.json > $tmp
+status=0
```

Comment:
I wish there was a way to just say `defer rm $tmp`.

Reply:
you technically can via a trap, but it's not particularly pretty either: http://redsymbol.net/articles/bash-exit-traps/